ludwig/data/preprocessing.py

```diff
@@ -316,17 +317,22 @@ def preprocess_for_training(
         preprocessing_params=default_preprocessing_parameters,
         random_seed=default_random_seed
 ):
-    # Check if hdf5 and json already exist
+    data_sources = [data_df, data_train_df, data_csv, data_train_csv,
+                    data_hdf5, data_train_hdf5]
+    data_sources_none = [x is None for x in data_sources]
+    if all(data_sources_none):
+        raise ValueError('No training data is provided!')
+
+    # Check if hdf5 and json already exist. If they do, use the hdf5 data,
+    # instead of the csvs
     data_hdf5_fp = None
-    data_train_hdf5_fp = None
-    data_validation_hdf5_fp = None
-    data_test_hdf5_fp = None
-    train_set_metadata_json_fp = 'metadata.json'
+
     if data_csv is not None:
-        data_hdf5_fp = os.path.splitext(data_csv)[0] + '.hdf5'
-        train_set_metadata_json_fp = os.path.splitext(data_csv)[0] + '.json'
-        if (os.path.isfile(data_hdf5_fp) and
-                os.path.isfile(train_set_metadata_json_fp)):
+        data_hdf5_fp = get_filepath_in_diff_format(data_csv, 'hdf5')
```

Comment:
would probably call this function `replace_extension` to make more generic

Reply:
sure. makes sense.

Reply:
done